### PR TITLE
Node 22 housekeeping

### PR DESCRIPTION
**Pull request by delucis**
## Changes

Small fix for type compatibility with Node 22. An optional second argument in [`fileURLToPath()`](https://nodejs.org/docs/latest-v22.x/api/url.html#urlfileurltopathurl-options) was added in Node 22.1.0 and our direct use with map meant we would be passing the array index as the options object.

<details>
<summary>Original comment</summary>

- #12540 updated GitHub workflows to use Node 22, but not the `.nvmrc` for local dev
- It also didn’t update `@types/node`. Updating to 22 caught an error due to the optional second argument in [`fileURLToPath()`](https://nodejs.org/docs/latest-v22.x/api/url.html#urlfileurltopathurl-options) added in Node 22.1.0, which this PR also fixes.
- The change to `@types/node` is a bit opinionated. I don’t know if we’d prefer to have this on 20 to avoid accidentally using APIs introduced in 22?

</summary>

## Testing
n/a existing tests should pass

## Docs

n/a

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-18.17.1
```
Comment (Princesseuh) [inline .nvmrc:1]: that should be our lowest supported version, not latest we support

File: package.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
     "@biomejs/biome": "1.9.3",
     "@changesets/changelog-github": "^0.5.0",
     "@changesets/cli": "^2.27.10",
-    "@types/node": "^18.17.8",
+    "@types/node": "^22.10.1",
```
Comment (Princesseuh) [inline package.json:61]: same here

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
...
+22.11.0
```
Comment (delucis) [inline .nvmrc:1]:
```suggestion
18.17.1
```

File: package.json (hunk 2 above)
Comment (delucis) [reply package.json:61]:
```suggestion
    "@types/node": "^18.17.8",
```

File: packages/telemetry/package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
   "devDependencies": {
     "@types/debug": "^4.1.12",
     "@types/dlv": "^1.1.5",
-    "@types/node": "^18.17.8",
+    "@types/node": "^22.10.1",
```
Comment (delucis) [inline packages/telemetry/package.json:43]:
```suggestion
    "@types/node": "^18.17.8",
```

**Review by ematipico — APPROVED** [review]

**Comment by delucis** [comment]
> why the type marks the usage wrong? (in Node v22)

They changed the signature in Node 22.1.0. Previously it was just `fileURLToPath(url: URL | string)`, but it was updated to add an optional options object as a second parameter: `fileURLToPath(url: URL | string, options?: { windows?: boolean })`.

But we’re using `@types/node` for Node 18 to ensure backwards compatibility so don’t get any type warning for this there.

**Pull request merged**